spec/Section 4 -- Introspection.md

```diff
@@ -69,33 +69,32 @@ underscores {"\_\_"}.
 ## Type Name Introspection
 
 GraphQL supports type name introspection within any selection set in an
-operation, with the single exception of selections at the root of a subscription
-operation. Type name introspection is accomplished via the meta-field
-`__typename: String!` on any Object, Interface, or Union. It returns the name of
-the concrete Object type at that point during execution.
+operation, with the single exception of a subscription _root selection set_.
```

Comment:
I believe this introduces ambiguity. 

A [selection set](https://spec.graphql.org/draft/#sec-Selection-Sets) consists of field, fragment spread and inline fragment selections. For example:

```graphql
subscription {
  ...Frag
}
fragment Frag on Subscription {
  ...Frag2
}
fragment Frag2 on Subscription {
  myField
}
```

^ this has **three** selection sets. The "root" selection set is, arguably, `{ ...Frag }`. (Note we also use the term "top level selection set" for this: https://spec.graphql.org/draft/#sel-LALPHDDDDDBCBIBf2we)

We need a term for a "resolved" selection set, i.e. the kind of thing you would get from `CollectFields()`. We call that `groupedFieldSet` in some places.

Perhaps the term "root field set" would work, but we'd also need to define the term "field set" I think.

Arguably "root field" is actually more correct here - there must be exactly one root field (after resolving the selection sets).

Reply:
Great find. Let me ponder on this

Reply:
Rephrasing this as "field collected from the root selection set"